diff --git a/samples/bluetooth/beacon/prj.conf b/samples/bluetooth/beacon/prj.conf
index 045c5c5f61d..dad48c73103 100644
--- a/samples/bluetooth/beacon/prj.conf
+++ b/samples/bluetooth/beacon/prj.conf
@@ -1,3 +1,5 @@
 CONFIG_BT=y
 CONFIG_LOG=y
 CONFIG_BT_DEVICE_NAME="Test beacon"
+CONFIG_DEBUG=y
+CONFIG_NO_OPTIMIZATIONS=y
\ No newline at end of file
diff --git a/samples/bluetooth/beacon/src/main.c b/samples/bluetooth/beacon/src/main.c
index 484b6e94c8f..1cedb95bdd2 100644
--- a/samples/bluetooth/beacon/src/main.c
+++ b/samples/bluetooth/beacon/src/main.c
@@ -40,6 +40,18 @@ static const struct bt_data sd[] = {
 	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
 };
 
+void mem_leak(){
+	printk("mem_leak()\n");
+	void *ptr = malloc(100);
+}
+
+void use_uninitialized_mem(){
+	printk("use_uninitialized_mem()\n");
+	int *ptr = malloc(sizeof(int));
+	int value = *ptr;
+	free(ptr);
+}
+
 static void bt_ready(int err)
 {
 	char addr_s[BT_ADDR_LE_STR_LEN];
@@ -80,6 +92,10 @@ int main(void)
 
 	printk("Starting Beacon Demo\n");
 
+	printk("Calling memleak functions\n");
+	mem_leak();
+	use_uninitialized_mem();
+
 	/* Initialize the Bluetooth Subsystem */
 	err = bt_enable(bt_ready);
 	if (err) {
