diff --git a/drivers/bluetooth/hci/userchan.c b/drivers/bluetooth/hci/userchan.c
index 50c3165809d..d0ac1e6d3c7 100644
--- a/drivers/bluetooth/hci/userchan.c
+++ b/drivers/bluetooth/hci/userchan.c
@@ -58,7 +58,7 @@ static K_KERNEL_STACK_DEFINE(rx_thread_stack,
 			     CONFIG_ARCH_POSIX_RECOMMENDED_STACK_SIZE);
 static struct k_thread rx_thread_data;
 
-static unsigned short bt_dev_index;
+static unsigned short bt_dev_index = 0;
 
 #define TCP_ADDR_BUFF_SIZE 16
 static bool hci_socket;
@@ -328,7 +328,8 @@ static int user_chan_open(void)
 	int fd;
 
 	if (hci_socket) {
-		struct sockaddr_hci addr;
+		struct sockaddr addr;
+		struct sockaddr_hci *hci_addr = (struct sockaddr_hci *)&addr;
 
 		fd = socket(PF_BLUETOOTH, SOCK_RAW | SOCK_CLOEXEC | SOCK_NONBLOCK,
 			    BTPROTO_HCI);
@@ -337,9 +338,9 @@ static int user_chan_open(void)
 		}
 
 		(void)memset(&addr, 0, sizeof(addr));
-		addr.hci_family = AF_BLUETOOTH;
-		addr.hci_dev = bt_dev_index;
-		addr.hci_channel = HCI_CHANNEL_USER;
+		hci_addr->hci_family = AF_BLUETOOTH;
+		hci_addr->hci_dev = bt_dev_index;
+		hci_addr->hci_channel = HCI_CHANNEL_USER;
 
 		if (bind(fd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
 			int err = -errno;
